src/wifi.rs

```diff
@@ -1355,3 +1358,110 @@ impl WifiWait {
         }
     }
 }
+
+#[cfg(all(feature = "async"))]
+pub struct EspAsyncWifiDriver<'d> {
+    sync: WifiDriver<'d>,
```

Comment:
Ideally, this should be `sync: T` `where T: BorrowMut<WifiDriver>`, so that the user has the freedom to pass a `&mut` ref as well. Also the driver you are wrapping here is NOT "sync" - see below, so you might want to change the field name.